src/safir/github/webhooks.py

```diff
@@ -0,0 +1,343 @@
+"""Pydantic models for GitHub webhook payloads."""
+
+from __future__ import annotations
+
+from enum import Enum
+
+from pydantic import BaseModel, Field
+
+from .models import (
+    GitHubCheckRunModel,
+    GitHubCheckSuiteModel,
+    GitHubPullRequestModel,
+    GitHubRepositoryModel,
+)
+
+__all__ = [
+    "GitHubAppInstallationModel",
+    "GitHubPushEventModel",
+    "GitHubAppInstallationEventRepoModel",
+    "GitHubAppInstallationEventAction",
+    "GitHubAppInstallationEventModel",
+    "GitHubAppInstallationRepositoriesEventAction",
+    "GitHubAppInstallationRepositoriesEventModel",
+    "GitHubPullRequestEventAction",
+    "GitHubPullRequestEventModel",
+    "GitHubCheckSuiteEventAction",
+    "GitHubCheckSuiteEventModel",
+    "GitHubCheckRunEventAction",
+    "GitHubCheckRunEventModel",
+]
+
+
+class GitHubAppInstallationModel(BaseModel):
+    """A Pydantic model for the ``installation`` field found in webhook
+    payloads for GitHub Apps.
+    """
+
+    id: str = Field(description="The installation ID.")
+
+
+class GitHubPushEventModel(BaseModel):
+    """A Pydantic model for the ``push`` event webhook when a commit or
+    tag is pushed.
+
+    https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads#push
+    """
+
+    repository: GitHubRepositoryModel = Field(
+        description="The repository that was pushed to."
+    )
+
+    installation: GitHubAppInstallationModel = Field(
+        description="Information about the GitHub App installation."
+    )
+
+    ref: str = Field(
+        description=(
+            "The full git ref that was pushed. Example: refs/heads/main or "
+            "refs/tags/v3.14.1."
+        ),
+        example="refs/heads/main",
+    )
+
+    before: str = Field(
+        description="The SHA of the most recent commit on ref before the push."
+    )
+
+    after: str = Field(
+        description="The SHA of the most recent commit on ref after the push."
+    )
+
+
+class GitHubAppInstallationEventRepoModel(BaseModel):
+    """A pydantic model for repository objects used by
+    `GitHubAppInstallationRepositoriesEventModel`.
+
+    https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads#installation
+    """
+
+    name: str = Field(
+        description="The name of the repository, e.g. 'times-square'."
+    )
+
+    full_name: str = Field(
+        description=(
+            "The full name of the repository, e.g. 'lsst-sqre/times-square'."
+        )
+    )
+
+    @property
+    def owner_name(self) -> str:
+        """The name of the repository owner."""
+        return self.full_name.split("/")[0]
+
+
+class GitHubAppInstallationEventAction(str, Enum):
+    """The action performed on an GitHub App ``installation`` webhook
+    (`GitHubAppInstallationEventModel`).
+    """
+
+    #: Someone installed a GitHub App on a user or organization account.
```

Comment:
Out of curiosity, how do you decide between this and the docstring syntax? Is one or the other better? Should we converge on one for our coding style? I've been using the docstring approach, but I'm happy to change.

Reply:
This was mostly a shot at getting enum documentation to work in the first place. I'll switch back to the docstring syntax to see if that works.